src/servant/tt_003/demon.c

```diff
@@ -240,7 +240,132 @@ INCLUDE_ASM("servant/tt_003/nonmatchings/demon", func_us_80172EF8);
+        prim->tpage = 0x1A;
+        prim->clut = 0x147;
+        prim->priority = self->zPriority + 1;
+        prim->drawMode = DRAW_TPAGE2 | DRAW_TPAGE | DRAW_COLORS | DRAW_TRANSP;
+        prim->u0 = prim->u2 = 0xE0;
+        prim->u1 = prim->u3 = 0xFF;
+        prim->v0 = prim->v1 = 0x80;
+        prim->v2 = prim->v3 = 0x9F;
+        PCOL(prim) = 0x80;
+
+        prim = prim->next;
+        prim->tpage = 0x1A;
+        prim->clut = 0x147;
+        prim->priority = self->zPriority + 1;
+        prim->drawMode =
+            DRAW_UNK_400 | DRAW_TPAGE2 | DRAW_TPAGE | DRAW_COLORS | DRAW_TRANSP;
+        prim->u0 = prim->u2 = 0;
+        prim->u1 = prim->u3 = 0x3F;
+        prim->v0 = prim->v1 = 0xC0;
+        prim->v2 = prim->v3 = 0xFF;
+
+        self->ext.et_801737F0.animationTriggerCount = 0;
+        self->ext.et_801737F0.animationTimer = 0;
+        self->ext.et_801737F0.angle = 0;
+
+        self->step++;
+        break;
+
+    case 1:
+        self->ext.et_801737F0.stepCounter++;
+        if ((self->ext.et_801737F0.stepCounter % 10 == 0) &&
+            (self->ext.et_801737F0.animationTriggerCount < 2)) {
+            self->ext.et_801737F0.animationTriggerCount++;
+        }
+
+        self->ext.et_801737F0.animationTimer += 8;
+
+        if (self->ext.et_801737F0.animationTimer >= 0x100) {
+            self->step++;
+        }
+        break;
+
+    case 2:
+        self->ext.et_801737F0.stepCounter++;
+
+        if (self->ext.et_801737F0.stepCounter > 50) {
+            CreateEventEntity(self, 0xDD, 0);
+            DestroyEntity(self);
+            return;
+        }
+        break;
+    }
+
+    self->ext.et_801737F0.angle += 0x400;
+    self->ext.et_801737F0.angle &= 0xFFF;
+
+    sine = rsin(self->ext.et_801737F0.angle);
+    cosine = rcos(self->ext.et_801737F0.angle);
+
+    posXOffset = (self->posX.i.hi) + ((self->facingLeft) ? (6) : (-6));
+    posYOffset = self->posY.i.hi - 0xC;
+    prim = &g_PrimBuf[self->primIndex];
+
+    scaledCounter = (self->ext.et_801737F0.animationTimer * 16) / 256;
+
+    prim->x0 = posXOffset +
+               ((cosine * -(scaledCounter)-sine * -(scaledCounter)) >> 12);
+    prim->y0 = posYOffset +
+               ((sine * -(scaledCounter) + cosine * -(scaledCounter)) >> 12);
+
+    prim->x1 =
+        posXOffset + ((cosine * (scaledCounter)-sine * -(scaledCounter)) >> 12);
+    prim->y1 = posYOffset +
+               ((sine * (scaledCounter) + cosine * -(scaledCounter)) >> 12);
+
+    prim->x2 =
+        posXOffset + ((cosine * -(scaledCounter)-sine * (scaledCounter)) >> 12);
+    prim->y2 = posYOffset +
+               ((sine * -(scaledCounter) + cosine * (scaledCounter)) >> 12);
+
+    prim->x3 =
+        posXOffset + ((cosine * (scaledCounter)-sine * (scaledCounter)) >> 12);
+    prim->y3 = posYOffset +
+               ((sine * (scaledCounter) + cosine * (scaledCounter)) >> 12);
+    newColor = ((self->ext.et_801737F0.stepCounter & 1) << 7);
+    PCOL(prim) = newColor;
```

Comment:
PSP match with
`PCOL(prim) = (self->ext.et_801737F0.animationTimer * 16) / 256;`

Then you can also get rid of the newColor var

Reply:
Not works on PSX - getting lbu instead of lhu instruction

Reply:
oh, I copied the wrong thing.  That was from a different function.  it should have been

`(self->ext.et_801737F0.stepCounter & 1) * 128`

That gets rid of the local var and does lhu.  Not a big deal.  I'm planning on onboarding this to the PSP build once we're done decomping and refactoring, so I'll catch it then.

Reply:
It works! I'll include your suggestions with next PR